rasa/server.py

```diff
@@ -889,3 +903,42 @@ def validate_request(rjs):
             )
 
     return app
+
+
+def _get_output_channel(
+    request: Request, tracker: Optional[DialogueStateTracker]
+) -> OutputChannel:
+    """Returns the `OutputChannel` which should be used for the bot's responses.
+
+    Args:
+        request: HTTP request whose query parameters can specify which `OutputChannel` 
+                 should be used.
+        tracker: Tracker for the conversation. Used to get the latest input channel.
+
+    Returns:
+        `OutputChannel` which should be used to return the bot's responses to.
+    """
+    requested_output_channel = request.args.get(OUTPUT_CHANNEL_QUERY_KEY)
+
+    if (
+        requested_output_channel == USE_LATEST_INPUT_CHANNEL_AS_OUTPUT_CHANNEL
+        and tracker
+    ):
+        requested_output_channel = tracker.get_latest_input_channel()
+
+    registered_input_channels = request.app.input_channels or []
+    matching_channels = [
+        channel
+        for channel in registered_input_channels
+        if channel.name() == requested_output_channel
+    ]
+
+    output_channel = reduce(
```

Comment:
Can you explain why we need this?

We get all registered channels and only keep the one that matches the requested output channel. So, isn't `len(matching_channels) <= 1`? Why reduce?

Reply:
(simplified this part of bit as part of working on my response to the comment)

the alternative would be 

```python
output_channel = None
if matching_channels:
   output_channel = matching_channels[0].get_output_channel()

return output_channel or CollectingOutputChannel()
```

with the `reduce` I can avoid the index access and it would handle it gracefully if the weird case happens and a channel name is in the list twice. Also the case where `get_output_channel()` returns `None` is handled neatly. But if you are not familiar with functional programming it might be harder to read 🤔

Reply:
I guess, it's fine. (It is just too warm to think 😄). But maybe you can just add a simple comment like `take the matching output channel or CollectingOutputChannel()`.

Reply:
Haha :-D Will do 👍